### ISSUE TITLE
word XIII < 32 bits

### ISSUE THREAD

**Issue by katieannemills** [GRIFFINCollaboration/data-spy#4]
according to the docs, type XIII words are 4 byte words consisting of:
- a 4 bit label (`0xE`)
- a 10 bit event trailer
- a 14 bit repeat of the low bits of the timestamp

Where are the other 4 bits?

**Comment by r3dunlop** [comment]
What we use in the sort code is
- 4 bit label (`0xE`)
- 14 bit Channel ID **NOT** Timestamp
- 2 bit NIOS code finish flag (where only one is used?)
- 8 bit NIOS code error
  So you are correct. I am unsure what the other 8 bits are being used for. Can @AdamGarnsworthy or @SmithJK comment?

**Comment by katieannemills** [comment]
what they're used for, or just where they are in the pattern.

Also @r3dunlop: missing bits aside, this doesn't match the spec; do the docs need to be updated? I can do with the rest of [the corrections](https://github.com/GRIFFINCollaboration/GRSISort/issues/604), if that's definitely what's happening. Moreover, are any of the other words different than what's described there?

**Comment by r3dunlop** [comment]
Here is the parsing code block:

``` c++
        case 0xe0000000:
            // changed on 21 Apr 2015 by JKS, when signal processing code from Chris changed the trailer.
            // change should be backward-compatible
            if((value & 0x3fff) == (EventFrag->ChannelId & 0x3fff)){
               if(!TGRSILoop::Get()->GetSuppressError() && EventFrag->DataType == 2) {
                  // check whether the nios finished and if so whether it finished with an error
                  if(((value>>14) & 0x1) == 0x1) {
                     if(((value>>16) & 0xff) != 0) {
                        printf( BLUE "0x%04x: NIOS code finished with error 0x%02x" RESET_COLOR "\n",EventFrag->ChannelAddress, (value>>16) & 0xff);
                     }
                  }
               }
```

So first 4 bits are the label, next 8 are ??? next 8 are NIOS code, next 2 are NIOS flag last 14 bits are channel Id. The current documentation is quite old now and I guess no one has gotten around to updating it. I can check to see if anything else is out of order. 

The other thing to be aware of is that there are PPG events hidden inside of GRIF events now. These have a different event format. I can lay that out in another issue for you.

**Comment by katieannemills** [comment]
Cool, thanks - so there's some ill-defined bits after the event label, got it.

PPG events specification: yes please!

**Comment by SmithJK** [comment]
This is going to change soon. There will be some kind of Accepted Channel Trigger ID here.

**Comment by SmithJK** [comment]
The NIOS errors should only come from the 4Gs, I think. @VinzenzBildstein, can you confirm this?

**Comment by r3dunlop** [comment]
If that’s true, then what are the rest of the bits in the trailer in the GRIF 16’s?

**Comment by SmithJK** [comment]
Good question. It's a shame Chris Pearson isn't on here. I'll send an email.

**Comment by r3dunlop** [comment]
Just add him as GRSIUser when he isn’t looking… :P

**Comment by VinzenzBildstein** [comment]
Yes, the NIOS errors were only implemented for the 4Gs (and only for some time), they will most likely be removed. I'm not sure yet what the final version of the trailer word will be for the 4Gs, but most likely it will be the same as the GRIF-16s.
